stream-chat-android-core/src/main/java/io/getstream/chat/android/models/streamcdn/image/extensions/StringExtensions.kt

```diff
@@ -0,0 +1,103 @@
+    return try {
+        val width = this.substringAfter("ow=", "").takeIf { it.isNotBlank() }
+            ?.substringBefore("&")?.toInt()
+        val height = this.substringAfter("oh=", "").takeIf { it.isNotBlank() }
+            ?.substringBefore("&")?.toInt()
+
+        if (height != null && width != null) {
+            StreamCdnOriginalImageDimensions(
+                originalWidth = width,
+                originalHeight = height
+            )
+        } else {
+            null
+        }
+    } catch (e: java.lang.Exception) {
+        val logger = StreamLog.getLogger("Chat: getStreamCDNHostedImageDimensions")
+        logger.e { "Failed to parse Stream CDN image dimensions from the URL:\n ${e.stackTraceToString()}" }
+
+        null
+    }
+}
+
+/**
+ * Generates a string URL with Stream CDN image resizing query parameters added to it. Once this URL is called, Stream's
+ * CDN will generate a resized image which is accessible using the link returned by this function.
+ *
+ * @param resizedWidthPercentage The percentage of the original image width the resized image width will be.
+ * @param resizedHeightPercentage The percentage of the original image height the resized image height will be.
+ */
+public fun String?.createResizedStreamCdnImageUrl(
+    @FloatRange(0.0, 1.0, fromInclusive = false) resizedWidthPercentage: Float,
+    @FloatRange(0.0, 1.0, fromInclusive = false) resizedHeightPercentage: Float,
+    resizeStrategy: StreamCdnResizeImageMode? = null,
+    cropMode: StreamCdnCropImageMode? = null
+): String? {
+    val streamCdnImageDimensions = this?.getStreamCdnHostedImageDimensions()
+
+    return if (this != null && streamCdnImageDimensions != null) {
+
+        val resizedWidth: Int = (streamCdnImageDimensions.originalWidth * resizedWidthPercentage).toInt()
+        val resizedHeight: Int = (streamCdnImageDimensions.originalHeight * resizedHeightPercentage).toInt()
+
+        this.appendValueAsQueryParameterIfNotNull(value = resizedWidth, name = "w")
+            .appendValueAsQueryParameterIfNotNull(value = resizedHeight, name = "h")
+            .appendValueAsQueryParameterIfNotNull(value = resizeStrategy?.queryParameterName, name = "resize")
+            .appendValueAsQueryParameterIfNotNull(value = cropMode?.queryParameterName, name = "crop")
+    } else {
+        val logger = StreamLog.getLogger("Chat:resizedStreamCdnImageUrl")
+        logger.w {
+            "Only images hosted by Stream's CDN containing original width and height query parameters" +
+                "can be resized"
+        }
+        this
+    }
+}
+
+/**
+ * A convenience method which evaluates if [value] is null or not and appends
+ * it with the accompanying parameter name in the form of a query parameter.
+ *
+ * @param value Query parameter value.
+ * @param name Query parameter value.
+ */
+private fun String.appendValueAsQueryParameterIfNotNull(value: Any?, name: String): String {
```

Comment:
Core is not an Android model so I couldn't use Android's `Uri`  here. Java's `URI` seemed more limited so I opted to using a custom string extension.

Open to suggestions here.

Reply:
Would be good to write some tests for this down the line, can be a future improvement